src/SqlStreamStore.Postgres/ActiveTransactions.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace SqlStreamStore
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Represents a collection of PostgreSQL transaction IDs (32-bit <c>xid</c>)
+    /// and provides logic for comparing 64-bit <c>xid8</c> snapshot IDs against
+    /// the currently active 32-bit <c>xid</c> values, taking into account
+    /// PostgreSQL’s circular wrap-around behavior (modulo 2^32).
+    /// </summary>
+    public class ActiveTransactions
+    {
+        private const ulong WRAP_BOUNDARY = 4294967296UL; // 2^32
+        private const uint HALF_BOUNDARY = 2147483648U; // 2^31
+
+        /// <summary>
+        /// Gets the list of active transactions as 32-bit <c>xid</c> values.
+        /// </summary>
+        public List<ActiveTransaction> TransactionIds { get; }
```

Comment:
You can't set this property but the `List` is still mutatable, maybe change the typing to hide it (or you can go for `ImmutableList` as well)
```suggestion
        public IEnumerable<ActiveTransaction> TransactionIds { get; }
```

Reply:
Changed it to IReadonlyList